### DISCUSSION TITLE
Understanding the meaning of the weights that we get with model.state_dict() and the method .predict()

### DISCUSSION THREAD

**Discussion by TiToLiOn** [lululxvi/deepxde#694] [Q&A]
Hi Lu,

I'm trying to reproduct the working of the method .predict() with a function, but I have found some differences between what I thought and what I see. I explain my dude with my code, it's just a simple example to show my dude.

<img width="553" alt="3" src="https://user-images.githubusercontent.com/97308944/169320040-d1e77c37-1539-4668-9527-779d8f09de74.PNG">

The structure of my net is 1  input layer with the 2 neurons variables of the pde, 2 hide layers with 5 neurons each one and 1 output layer with 1 neuron which is the value predicted of my net. 

When I execute model.state_dict() I get what I expected in part, butI receive more information that I don't know how to use.

First of all I receive in order, weights and bias of each of the layers of my net

<img width="566" alt="w1" src="https://user-images.githubusercontent.com/97308944/169320047-f3b13656-51ad-414e-a841-aedfbdc5a410.PNG">

But apart from that, I got parameters called beta1 power, beta2_power and weights and bias of adam that I don't know how they matter in the prediction.

<img width="488" alt="w2" src="https://user-images.githubusercontent.com/97308944/169320029-404ce430-5b8e-4e8a-a6e9-e5eb254dc0b3.PNG">
<img width="504" alt="w3" src="https://user-images.githubusercontent.com/97308944/169320037-4733b053-f4cb-492c-bd96-52bce8557e1f.PNG">

My function trying to replicate the method .predict is 
<img width="303" alt="p" src="https://user-images.githubusercontent.com/97308944/169320045-2a14183e-2d75-4d67-98b3-d788e738ce87.PNG">

It operates with the matrix of the weights and bias just of my net and composing with activation function, but ignores the rest of parametres, and I though that this was the normal way to predict. Could you explain me the meaning of the parametres beta1 power, beta2_power and weights and bias of adam, and how does the .predict() method works.

Thank you a lot.

**Comment by lululxvi** [comment]
Check the Adam paper for those variables.
